examples/chip-tool/commands/clusters/ClusterCommand.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <app/CommandSender.h>
+
+#include "DataModelLogger.h"
+#include "ModelCommand.h"
+
+class ClusterCommand : public ModelCommand, public chip::app::CommandSender::Callback
+{
+public:
+    ClusterCommand(const char * commandName) : ModelCommand(commandName)
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+    }
+
+    ~ClusterCommand() {}
+
+    /////////// CommandSender Callback Interface /////////
+    virtual void OnResponse(chip::app::CommandSender * client, const chip::app::ConcreteCommandPath & path,
+                            const chip::app::StatusIB & status, chip::TLV::TLVReader * data) override
+    {
+        if (CHIP_NO_ERROR != status.ToChipError())
+        {
+            ChipLogError(chipTool, "Response Failure: %s", chip::ErrorStr(status.ToChipError()));
+            SetCommandExitStatus(status.ToChipError());
+            return;
+        }
+
+        if (data != nullptr)
+        {
+            CHIP_ERROR error = DataModelLogger::LogCommand(path, data);
+            if (CHIP_NO_ERROR != error)
+            {
+                ChipLogError(chipTool, "Response Failure: Can not decode Data");
+                SetCommandExitStatus(error);
+                return;
+            }
+        }
+    }
+
+    virtual void OnError(const chip::app::CommandSender * client, CHIP_ERROR error) override
+    {
+        ChipLogProgress(chipTool, "Internal Error: %s", chip::ErrorStr(error));
```

Comment:
This could also be a non-path-specific error response from the server, so not sure about this labeling.  Similar for the other files.